docs/guides/file-system-based-automated-bundle-generation.md

```diff
@@ -0,0 +1,133 @@
+
+### Configure Components Directory
+`components_directory` is the directory used to automatically detect and register components for usage on the rails view.
+Configure `config/initializers/react_on_rails`
+to set the name for `components_directory`. 
+
+```rb
+config.components_directory = "ror_components"
+```
+
+Now all React components inside `config.components_directory` will automatically be registered for the usage with `react_component` and `react_component_hash` helper methods provided by React on Rails.
+
+### Configure `auto_load_bundle` Option
+
+For automated component registry, `react_component` and `react_component_hash` view helper method tries to load generated bundle for component from the generated directory automatically per `auto_load_bundle` option. `auto_load_bundle` option in `config/initializers/react_on_rails` configures the default value that will be passed to component helpers. The default is `false`, and the option can be passed at the time of individual usage too.
+
+*Note: Starting from React on Rails version 15.0.0, the default value for `auto_load_bundle` option will be `true`.*
+
+You can change the value in `config/initializers/react_on_rails` by updating as follows:
+
+```rb
+config.auto_load_bundle = true
+```
+
+### Update `.gitignore` file
+React on Rails, per the settings in your `config/webpacker.yml`, automatically generates pack files for components to be registered in the `packs/generated` directory. To avoid committing generated files into the version control system, Please update `.gitignore` to have 
+
+```gitignore
+### Generated React on Rails packs
+packs/generated
+```
+
+*Note: the directory might be different depending on the `source_entry_path` in `config/webpacker.yml`.*
+
+## Usage
+
+### Basic usage
+
+if the `webpacker.yml` file is configured as instructed [here](https://github.com/shakacode/shakapacker#configuration-and-code), With following configurations
+
+```yml
+default: &default
+    source_path: app/javascript
+    source_entry_path: packs 
+    public_root_path: public
+    public_output_path: packs
+    nested_entries: true
+# And more 
+```
+
+the directory structure will look like this
+```
+app/javascript:
+  └── packs:               # sets up webpack entries
+  │   └── application.js   # references ../src/my_component.js
+  │   └── application.css
+  └── src:                 # any directory name is fine. Referenced files need to be under source_path
+  │   └── component.js
+  └── stylesheets:
+  │   └── my_styles.css
+  └── images:
+      └── logo.svg
+public/packs                # webpack output
+```
+
+Now, To automatically register, `A_ComponentOne`, `B_ComponentOne`, `B_ComponentTwo` for the usage with `react_component` and `react_component_hash` helpers, Create directory structure as mentioned below:
+
+```
+app/javascript:
+  └── packs:                   
+  │   └── application.js     
+  │   └── application.css
+  └── src:                   
+  │   └── A
```

Comment:
What are A and B here? Should they be in the above directory structure too?

Reply:
@pulkitkkr This is the last point to address, please consider it approved otherwise! 🙌